plugins/modules/cloud/misc/terraform.py

```diff
@@ -459,6 +471,12 @@ def main():
         for f in variables_files:
             variables_args.extend(['-var-file', f])
 
+    if declare_variables:
+        gen_variables_file = tempfile.NamedTemporaryFile(mode='w', dir=project_path, prefix="generated_variables_", suffix=".tf")
```

Comment:
Hi russoz,

I initially wrote this code that way (adding the generated file to the `variable_files`) and it did not work. Apparently the `variables_files` only accepts `.tfvars` files (which can supply the values of variables with lines of the format `key=value`, but cannot declare variables). The declaration lines I am generating (of the format `variable key {}`) can only go in a regular `.tf` file not a `.tfvars`. Perhaps someone with more Terraform knowledge can think of a cleaner way, but the only way I could make it work was by putting it in the project path since that's where it looks for `.tf` files.

Reply:
Alright, thanks for the clarification!